### PR TITLE
Add functionality in MonsterData

### PR DESCRIPTION
**Pull request by ChisatoNishikigi73**
## Description
Add some functionality in MonsterData.java

stopAi(This only seems to change the state at the time of generation)
setKillDropId
setSkin
setDescribeId
setBaseHp
setBaseAttack
setBaseDefense
setElecSubHurt
setGrassSubHurt
setWaterSubHurt
setWindSubHurt
setIceSubHurt
setPhysicalSubHurt
setCampID
setWeaponId

## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by Hartie95** [comment]
You could probably use the @Getter and @Setter annotations for adding the setters and moving the getter code out there.

**Comment by Hartie95** [comment]
Also might not be a good idea to directly edit the resources data objects, instead it might be better to use values from there as base for new properties on the game entities and edit them.

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+    private int id;
+
+    private String monsterName;
     private MonsterType type;
     private String serverScript;
     private List<Integer> affix;
     private String ai;
     private int[] equips;
-    private List<HpDrops> hpDrops;
+    private List<emu.grasscutter.data.excels.MonsterData.HpDrops> hpDrops;
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/data/excels/MonsterData.java:22]: Why?

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java [hunk 2]
```diff
@@ -42,159 +42,222 @@ public class MonsterData extends GameResource {
...
-	        return this.HpPercent;
-	    }
-	}
+
+    @Override
+    public int getId() {
+        return this.id;
+    }
+
+    public String getMonsterName() {
```
Comment (Birdulon) [inline src/main/java/emu/grasscutter/data/excels/MonsterData.java:55]: Just use Lombok `@Getter` and `@Setter`?

**Review by Birdulon — CHANGES_REQUESTED** [review]
For what purpose?

File: src/main/java/emu/grasscutter/data/excels/MonsterData.java (hunk 1 above)
Comment (ChisatoNishikigi73) [reply src/main/java/emu/grasscutter/data/excels/MonsterData.java:22]: I will change it. It's my ide's problem.

**Review by GanyusLeftHorn — COMMENTED** [review]
Excel data classes are meant to represent static resources that we read when the server starts, and never touch again. In other words, they are immutable. They have no setters by design, not by accident. (Aside: The only reason why their fields aren't declared `final`, or why we don't use `record`s for them, is that GSON seems to have a problem with that.)

As such, I would reject this PR.

**Comment by ChisatoNishikigi73** [comment]
> For what purpose?

@ GanyusLeftHorn :
Suggestion: Maybe we could also add the ability to spawn monsters (1) with a set amount of HP, which might be very large/infinite, (2) with the AI turned off. This would effectively make it possible to spawn a test dummy monster.

Something like that would have been very useful while working on energy, for example, and would also be useful for testing if certain passives etc. work.

I just want to learn the language better, and make a contribution to Grasscutter, if I can help others, nothing else

**Comment by GanyusLeftHorn** [comment]
> I just want to learn the language better, and make a contribution to Grasscutter, if I can help others, nothing else

@Kawaa-qwq That's great! Contributions are very welcome. Please don't be discouraged if you get negative feedback to one of your PRs. 

If you are unsure about what exactly to contribute, or whether your approach for a particular contribution is good, you can come discuss it with us in the #development channel in the Discord.

**Pull request closed without merging**